### ISSUE TITLE
Display warning about hook in .overcommit.yml without explicit `enabled: true`

### ISSUE THREAD

**Issue by sds** [brigade/overcommit#173]
While converting our existing configurations to include all the hooks that are now disabled by default, it was noticed that having to explicitly include `enabled: true` was a bit of a nuisance, considering it was not previously necessary for most hooks (but now that https://github.com/brigade/overcommit/issues/122#issuecomment-87114806 is implemented, an explicit `enabled: true` is necessary for all hooks included in `.overcommit.yml`).

One suggestion was that we could just default to `enabled: true` for hooks that appear in `.overcommit.yml`, as this is a strong-enough signal that the user wants to enable the hook. However, since most hooks will probably not require any custom configuration, this would result in configurations looking like:

```yaml
PreCommit:
  HookOne: {}

  HookTwo: {}
```

...which is a bit odd to read, and the requirement of an explicit `{}` is a quirk of YAML that most would not realize. Thus it seems like consistently requiring `enabled: true` would read better and serve as a reminder that each hook can be configured, and also result in cleaner diffs when adding/removing options (since you don't need to touch the `HookOne:` line).

```yaml
PreCommit:
  HookOne:
    enabled: true

  HookTwo:
    enabled: true
```

Since we're pushing for `enabled` to be explicitly required, we should display a warning (or just fail with a configuration error) if a hook that is present in `.overcommit.yml` does not have an explicit `enabled` option listed, in order to help new users get started with the tool. If they do not intend for the hook to be enabled, they can always set it to `enabled: false`.

Blocker for 0.24.0 release. Comments welcome.

**Comment by lencioni** [comment]
I am definitely in favor of keeping `enabled` explicit. I think this will be easier to understand. And, as you outlined we should display a warning if the hook is present and does not explicitly state `enabled`ness. :+1:

**Comment by jawshooah** [comment]
Agreed with the above, and I'd side with just displaying a warning rather
than erroring out, since the config is still usable.

**Comment by sds** [comment]
Implemented warning display in 01e6cab7f6f56f2d8990c7642f80d316a2fbe3df.
